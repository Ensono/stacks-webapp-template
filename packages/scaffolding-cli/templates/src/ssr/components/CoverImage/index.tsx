import {PrefixedLink as Link} from "components"
import React from "react"

type CoverImageType = {
    title: string
    url: string
    slug?: string
}

export default function CoverImage({title, url, slug}: CoverImageType) {
    const image = <img src={url} alt={`Cover Image for ${title}`} />
    return (
        <div>
            {slug ? (
                <Link as={`/posts/${slug}`} href="/posts/[...all]">
                    <a aria-label={title}>{image}</a>
                </Link>
            ) : (
                image
            )}
        </div>
    )
}
