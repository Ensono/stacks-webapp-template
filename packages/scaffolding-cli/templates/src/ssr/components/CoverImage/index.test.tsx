import {render} from "@testing-library/react"
import React from "react"
import CoverImage from "."

test("renders ApiPane snapshot", () => {
    const {asFragment} = render(<CoverImage title="" url="" slug="" />)
    expect(asFragment()).toMatchSnapshot()
})
