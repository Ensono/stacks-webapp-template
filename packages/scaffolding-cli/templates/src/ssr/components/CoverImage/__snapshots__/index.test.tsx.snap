// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ApiPane snapshot 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="Cover Image for "
      src=""
    />
  </div>
</DocumentFragment>
`;
